src/api/app/models/repository_architecture.rb

```diff
@@ -1,11 +1,26 @@
 class RepositoryArchitecture < ApplicationRecord
+  serialize :required_checks, Array
   belongs_to :repository,   inverse_of: :repository_architectures
   belongs_to :architecture, inverse_of: :repository_architectures
 
   acts_as_list scope: [:repository_id], top_of_list: 0
 
   validates :repository, :architecture, :position, presence: true
   validates :repository, uniqueness: { scope: :architecture }
+
+  has_many :status_reports, as: :checkable, class_name: 'Status::Report', dependent: :destroy do
+    def for_uuid(uuid)
+      where(status_reports: { uuid: uuid })
+    end
+
+    def latest
+      for_uuid(proxy_association.owner.build_id).first
+    end
+  end
```

Comment:
This is basically c&p from the Repository? Should we move it to a module?

Reply:
I guess. One more concern :(

Reply:
Are you concerned :thinking:

Reply:
well, I'm with https://medium.com/@carlescliment/about-rails-concerns-a6b2f1776d7d on this one